### PR TITLE
Runs the webhook authorizer as a daemon on the guest machine / k8s master.

### PR DESCRIPTION
**Pull request by filmil**
With the future of the webhook auth in flux (see
https://github.com/kubernetes/kubernetes/issues/52511) it is not quite
certain what is to become of the final webhook authz interface.

But we can't wait until that happens - we have a job to do.  This CL
brute-forces the minikube setup to run a webhook authorizer on the same
machine as the apiserver.  This makes it trivial for the apiserver to
reach the authorizer, at the expense of some inflexibility and
non-scalability.

This is a bet that webhook authz will be resolved favorably until we get
to building the actual service based on this prototype.

The authorizer is deployed through the `addons` directory in minikube
home, and is activated by injecting a script that makes the program file
executable and then runs it.

The changes required for this trick to work:

- gencert.sh: modified to generate certs for CN=localhost, since this
is what is going to be used for the webhook URL.

- main.go: in order for the daemon to be startable via systemd, we need
to write to a inotify Unix socket.

- minikube/init.sh: this script copies all the files that are needed
for the setup to work.

- minikube/bootlocal.sh: this script will be called post-startup to
activate the authorizer.

- minikube/start.sh: This script injects the bootlocal.sh and runs it
on the remote machine.

---
This change is [Reviewable](https://reviewable.io/reviews/mdruskin/kubernetes-enterprise-control/15)

### CODE REVIEW

**Comment by filmil** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/15)*

**Comment by filmil** [comment]
Folks, ping?

File: cmd/authorizer/minikube/start.sh [hunk 1]
```diff
@@ -13,3 +13,13 @@ minikube start \
 
 minikube ip
 
+# Injects and makes executable the script that will run the authorizer on
+# the minikube master.
+BOOTLOCAL=$(cat minikube/bootlocal.sh)
+minikube ssh -- "cat > bootlocal.sh << EOF \
```
Comment (briantkennedy) [inline cmd/authorizer/minikube/start.sh]: Is it possible to put this in ~/.minikube/addons like the other key/cert?

**Comment by filmil** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[cmd/authorizer/minikube/start.sh, line 19 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/15#-Ku5ONQ2Ytork9JMamK3-r1-19:-KuKl1oy5j50vI5SVuIq:b4otgve) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/769659414d54df30a565457ee15b94857f69bdcd/cmd/authorizer/minikube/start.sh#L19)):*
<details><summary><i>Previously, briantkennedy (Brian Kennedy) wrote…</i></summary><blockquote>

Is it possible to put this in ~/.minikube/addons like the other key/cert?
</blockquote></details>

Yes indeed.  Done.  

Good catch.

For the record, the injection was from the time I was trying to  add the required setup in `/var/lib/boot2docker/bootlocal.sh`.  That turned out not to be feasible since at the time `bootlocal.sh` was run, the contents of the `addons` directory were not yet copied over.  But I then forgot to remove the injection.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/15)*

**Comment by filmil** [comment]
PTAL?

I had to change a few things, which seems to have undone your LGTM.

**Comment by filmil** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[cmd/authorizer/minikube/start.sh, line 19 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/15#-Ku5ONQ2Ytork9JMamK3-r1-19:-KuLH6laFO2cqDS86OC8:bb6ywmx) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/769659414d54df30a565457ee15b94857f69bdcd/cmd/authorizer/minikube/start.sh#L19)):*
<details><summary><i>Previously, filmil (Filip Filmar) wrote…</i></summary><blockquote>

Yes indeed.  Done.  

Good catch.

For the record, the injection was from the time I was trying to  add the required setup in `/var/lib/boot2docker/bootlocal.sh`.  That turned out not to be feasible since at the time `bootlocal.sh` was run, the contents of the `addons` directory were not yet copied over.  But I then forgot to remove the injection.


</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/15)*

**Pull request merged**